### PR TITLE
Docs: Document all metadata tables.

### LINKED ISSUE #757: Document all metadata tables
We should document our entries metadata table that shows all manifest entries.

### PR DESCRIPTION
**Pull request by nk1506**
added missing metadata tables  `all_delete_files` `all_entries` `entries` `position_deletes` 

Fixes #757

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
@nk1506: I think it is hard to review. Can you please keep the scope of this PR to add the missing metadata tables? Rearranging can be done in a followup PR.

File: docs/spark-queries.md [hunk 1]
```diff
@@ -335,6 +347,18 @@ Note:
 
 2. The partitions metadata table shows partitions with data files or delete files in the current snapshot. However, delete files are not applied, and so in some cases partitions may be shown even though all their data rows are marked deleted by delete files.
 
+### Positional Delete Files
+
+To show all positional delete files from a table:
```
Comment (ajantha-bhat) [inline docs/spark-queries.md:352]: only from current snapshot? If it is not referenced by current snapshot, it won't show right? I think we need to clarify

File: docs/spark-queries.md [hunk 2]
```diff
@@ -357,6 +381,31 @@ SELECT * FROM prod.db.table.all_data_files;
 |      0|s3://.../dt=20210103/00000-0-26222098-032f-472b-8ea5-651a55b21210-00001.parquet|    PARQUET|{20210103}|          14|              2444|{1 -> 94, 2 -> 17}|{1 -> 14, 2 -> 14}|  {1 -> 0, 2 -> 0}|              {}|{1 -> 1, 2 -> 20210103}|{1 -> 3, 2 -> 20210103}|        null|          [4]|        null|            0|
 |      0|s3://.../dt=20210104/00000-0-a3bb1927-88eb-4f1c-bc6e-19076b0d952e-00001.parquet|    PARQUET|{20210104}|          14|              2444|{1 -> 94, 2 -> 17}|{1 -> 14, 2 -> 14}|  {1 -> 0, 2 -> 0}|              {}|{1 -> 1, 2 -> 20210104}|{1 -> 3, 2 -> 20210104}|        null|          [4]|        null|            0|
 
+#### All Delete Files
+
+To show all the table's delete files and each file's metadata:
```
Comment (ajantha-bhat) [inline docs/spark-queries.md:386]: same comment for "all the table's" 

do we need each file's metadata?

File: docs/spark-queries.md [hunk 3]
```diff
@@ -357,6 +381,31 @@ SELECT * FROM prod.db.table.all_data_files;
...
+| 2 | s3:/.../table/data/00047-25-833044d0-127b-415c-b874-038a4f978c29-00612.parquet | PARQUET | 0 | 126506 | 28613985 | {100:135377,101:11314} | {100:126506,101:126506} | {100:105434,101:11} | {} | {100:0,101:17} | {100:404455227527,101:23} | NULL | NULL | [1] | 0 | {"id":{"column_size":135377,"value_count":126506,"null_value_count":105434,"nan_value_count":null,"lower_bound":0,"upper_bound":404455227527},"data":{"column_size":11314,"value_count":126506,"null_value_count": 11,"nan_value_count":null,"lower_bound":17,"upper_bound":23}} |
+
+#### All Entries
+
+To show all the table's manifest entries from any reachable snapshot for both data and delete files:
```
Comment (ajantha-bhat) [inline docs/spark-queries.md:399]: from any reachable snapshot -> from all the snapshots

Comment (ajantha-bhat) [reply docs/spark-queries.md:399]: also " all the table's" is confusing here too.

File: docs/spark-queries.md (hunk 2 above)
Comment (nk1506) [reply docs/spark-queries.md]: followed the same convention of previous tables like `files`

Comment (ajantha-bhat) [reply docs/spark-queries.md]: ok. The existing convention is still confusing for me to read. But we can optimize in a follow up. Ok for me to keep it similar now.

**Comment by nk1506** [comment]
@szehon-ho , Please review and share the feedback.

**Review by nastra — APPROVED** [review]

**Review by ajantha-bhat — APPROVED** [review]
LGTM

**Pull request merged**